marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.mjs

```diff
@@ -67,10 +67,30 @@ export function getArtifacts(artifactType) {
 export function deleteArtifact(artifactType, artifactName, artifactVersion = 'latest') {
     const artifactKey = generateArtifactKey(artifactType, artifactName, artifactVersion);
     const artifactLibrary =  getArtifactTypeLibrary(artifactType);
+
     // Currently there is no versioning for loadData artifacts
     const node = getArtifactNode(artifactType, artifactName, artifactVersion);
+
+    let settingNode = {};
+    if (artifactType == 'loadData') {
+        //delete related config file if existed
+        settingNode = artifactLibrary.getArtifactSettingNode(artifactName,
```

Comment:
Is the issue that loadData has a separate settings document, but other artifacts do not? If so, why is it a separate document? Conceptually, I would think that the "settings" of a "loadData" would be part of the "loadData" document.

Reply:
@rjrudin The implementation was based on https://project.marklogic.com/jira/browse/DHFPROD-4096
I assumed we could stored them as two different kinds of documents for flexibility and easier write/read. @ryanjdew What do you think?

Reply:
I don't see anything in 4096 that suggests any benefits from having a separate document. The middle tier of course won't care. But in ML, I think life is a lot simpler if it's a single document. You may need a separate endpoint for just changing settings on loadData specifically, as opposed to trying to toss that into the generic artifact endpoint. That would be good in the sense that the generic artifact endpoint wouldn't end up with multiple "If this is loadData, then do this loadData-specific code".

Reply:
https://project.marklogic.com/jira/browse/DHFPROD-4173
Data mapping settings only have three more properties than data loading settings (Source database,  Provenance granularity and Target Format). I think we should consider them...

Reply:
I personally like the separate documents. We're treating the writing of the settings and the artifact separately in the UI. Also, the settings serve a different purpose. With the exception of ingest/load-data, the artifacts themselves will be primarily used server-side, while the settings will be used by the Data Hub orchestration code.